apps/src/studio/studio.js

```diff
@@ -3111,17 +3234,64 @@ Studio.setWalls = function (opts) {
   }
 
   if (opts.value === Studio.walls) {
+    Studio.fixSpriteLocation();
     return;
   }
 
   Studio.walls = opts.value;
 
-  // Draw the tiles (again) that we know which background we're using.
+  // Draw the tiles (again) now that we know which background we're using.
   $(".tile_clip").remove();
   $(".tile").remove();
   Studio.drawMapTiles();
+
+  Studio.fixSpriteLocation();
+
+  sortDrawOrder();
+};
+
+/**
+ * A call to setWalls might place a wall on top of the sprite.  In that case,
+ * find a new nearby location for the sprite that doesn't have a wall.
+ */
+Studio.fixSpriteLocation = function () {
+  if (level.wallMapCollisions && level.blockMovingIntoWalls) {
+    var spriteIndex = 0;
+    var sprite = Studio.sprite[spriteIndex];
+    var xPos = getNextPosition(spriteIndex, false, false);
+    var yPos = getNextPosition(spriteIndex, true, false);
+
+    if (Studio.willSpriteTouchWall(sprite, xPos, yPos)) {
+
+      // Let's assume that one of the surrounding 8 squares is available.
+
+      var xCenter = xPos + sprite.width / 2;
+      var yCenter = yPos + sprite.height / 2;
+
+      var xGrid = Math.floor(xCenter / Studio.SQUARE_SIZE);
+      var yGrid = Math.floor(yCenter / Studio.SQUARE_SIZE);
+
+      var minRow = Math.max(yGrid - 1, 0);
+      var maxRow = Math.min(yGrid + 1, Studio.ROWS - 1);
+      var minCol = Math.max(xGrid - 1, 0);
+      var maxCol = Math.min(xGrid + 1, Studio.COLS - 1);
+
+      for (var row = minRow; row <= maxRow; row++) {
+        for (var col = minCol; col <= maxCol; col++) {
+          if (! Studio.isWallTile(row, col)) {
+            sprite.x = Studio.HALF_SQUARE + Studio.SQUARE_SIZE * col - sprite.width / 2;
```

Comment:
Any reason you can't just do `Studio.SQUARE_SIZE * col`? I guess the sprite sizes are diverging from the grid sizes, so perhaps there is an issue with large sprites?